pkg/log/io/io.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package io
+
+import (
+	"log/slog"
+	"os"
+)
+
+// IOController manages the input and output for the CLI
+type IOController struct {
+	in     *InputController
+	out    *OutputController
+	logger *oktetoLogger
+}
+
+// NewIOController returns a new input/output controller
+func NewIOController() *IOController {
+	ioController := &IOController{
+		out:    newOutputController(os.Stdout),
+		in:     newInputController(os.Stdin),
+		logger: newOktetoLogger(),
+	}
+	return ioController
+}
+
+// In returns the input reader
+func (ioc *IOController) In() *InputController {
+	return ioc.in
+}
+
+// Out returns the output writer
+func (ioc *IOController) Out() *OutputController {
+	return ioc.out
+}
+
+// Logger returns the logger
+func (ioc *IOController) Logger() *slog.Logger {
+	return ioc.logger.Logger
+}
+
+func (ioc *IOController) SetLevel(lvl string) {
```

Comment:
if this is part of Logger.... should it be under Logger domain? instead of IOController?

Reply:
Resolved